src/parseRuntimeAndValidateSDK.ts

```diff
@@ -70,40 +82,60 @@ export function getHumanFriendlyRuntimeName(runtime: string): string {
   return _.get(MESSAGE_FRIENDLY_RUNTIMES, runtime, runtime);
 }
 
-/**
- * Returns the Node.js version to be used for the function(s) as defined in the
- * package.json.
- * @param sourceDir directory where the functions are defined.
- * @return The runtime, e.g. `nodejs10`.
- */
-export function getRuntimeChoice(sourceDir: string): string {
+function getRuntimeChoiceFromFirebaseJson(runtimeFromConfig: string): string {
+  if (!ENGINE_RUNTIMES_NAMES.includes(runtimeFromConfig)) {
+    track("functions_runtime_notices", "package_missing_runtime");
+    throw new FirebaseError(UNSUPPORTED_NODE_VERSION_FIREBASE_JSON_MSG, { exit: 1 });
+  }
+
+  if (runtimeFromConfig === "nodejs6") {
+    track("functions_runtime_notices", "nodejs6_deploy_prohibited");
+    throw new FirebaseError(UNSUPPORTED_NODE_VERSION_FIREBASE_JSON_MSG, { exit: 1 });
+  }
+
+  return runtimeFromConfig;
+}
+
+function getRuntimeChoiceFromPackageJson(sourceDir: string): string {
   const packageJsonPath = path.join(sourceDir, "package.json");
   const loaded = cjson.load(packageJsonPath);
   const engines = loaded.engines;
   if (!engines || !engines.node) {
-    // We should really never hit this, since deploy/functions/prepare already checked that package.json has an "engines" field.
-    throw new FirebaseError(
-      `Engines field is required but was not found in package.json.\n` +
-        `To fix this, add the following lines to your package.json: \n
-      "engines": {
-        "node": "10"
-      }\n`
-    );
+    // We should really never hit this, since deploy/functions/prepare already checked that
+    // the runtime is defined in either firebase.json or the "engines" field of the package.json.
+    throw new FirebaseError(RUNTIME_NOT_SET);
   }
+
   const runtime = ENGINE_RUNTIMES[engines.node];
   if (!runtime) {
     track("functions_runtime_notices", "package_missing_runtime");
-    throw new FirebaseError(UNSUPPORTED_NODE_VERSION_MSG, { exit: 1 });
+    throw new FirebaseError(UNSUPPORTED_NODE_VERSION_PACKAGE_JSON_MSG, { exit: 1 });
   }
 
   if (runtime === "nodejs6") {
     track("functions_runtime_notices", "nodejs6_deploy_prohibited");
-    throw new FirebaseError(UNSUPPORTED_NODE_VERSION_MSG, { exit: 1 });
-  } else {
-    if (functionsSDKTooOld(sourceDir, ">=2")) {
-      track("functions_runtime_notices", "functions_sdk_too_old");
-      utils.logWarning(FUNCTIONS_SDK_VERSION_TOO_OLD_WARNING);
-    }
+    throw new FirebaseError(UNSUPPORTED_NODE_VERSION_PACKAGE_JSON_MSG, { exit: 1 });
   }
+
+  return runtime;
+}
+
+/**
+ * Returns the Node.js version to be used for the function(s) as defined in the
+ * either the `runtime` field of the rc file or the package.json.
```

Comment:
I think you mean "firebase.json file", not "rc file".

Reply:
Fixed